src/SignalR/clients/csharp/Http.Connections.Client/src/Internal/WebSocketsTransport.cs

```diff
@@ -118,6 +119,34 @@ private async ValueTask<WebSocket> DefaultWebSocketFactory(WebSocketConnectionCo
                 }
 
                 context.Options.WebSocketConfiguration?.Invoke(webSocket.Options);
+
+#if NET7_0_OR_GREATER
+                if (webSocket.Options.HttpVersion >= HttpVersion.Version20)
+                {
+                    // Reset options we set on the users' behalf since they are already on the HttpClient that we're passing to ConnectAsync
+                    // And ConnectAsync will throw if these options are set on the ClientWebSocketOptions
+                    if (ReferenceEquals(webSocket.Options.Cookies, context.Options.Cookies))
+                    {
+                        webSocket.Options.Cookies = null;
+                    }
+                    if (webSocket.Options.ClientCertificates == context.Options.ClientCertificates)
+                    {
+                        webSocket.Options.ClientCertificates.Clear();
```

Comment:
Is this a clone? That makes sense, but then will `webSocket.Options.ClientCertificates == context.Options.ClientCertificates` ever be true then? Does `X509CertificateCollection` define a non-default `==` operator?

Reply:
I think we have to loop over the certificates and reference equal the individual certs. Updated.